### PR TITLE
[MM-52845]: Migrate "components/admin_console/elasticsearch_settings.jsx" and tests to Typescript

### PR DESCRIPTION
**Pull request by karan2704**
#### Summary

Migrate components/admin_console/elasticsearch_settings.jsx to typescript

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/23415
Jira https://mattermost.atlassian.net/browse/MM-52845

#### Screenshots

N/A

#### Release Note

```release-note
N/A

```

### CODE REVIEW

**Comment by karan2704** [comment]
kindly go through the changes... I'm still working on the tests and I'll add them once done

File: webapp/channels/src/components/admin_console/elasticsearch_settings.tsx [hunk 1]
```diff
@@ -218,6 +238,7 @@ export default class ElasticsearchSettings extends AdminSettings {
                         />
                     }
                     placeholder={Utils.localizeMessage('admin.elasticsearch.caExample', 'E.g.: "./elasticsearch/ca.pem"')}
+                    type='input'
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/elasticsearch_settings.tsx:241]: type is optional and its default is input

File: webapp/channels/src/components/admin_console/jobs/table.tsx [hunk 2]
```diff
@@ -21,7 +21,7 @@ import './table.scss';
 
 export type Props = {
     jobs: Job[];
-    getExtraInfoText?: (job: Job) => string;
+    getExtraInfoText?: (job: Job) => React.ReactElement|null;
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/jobs/table.tsx:24]: use ReactNode to undertake null as well in the type

File: webapp/channels/src/components/admin_console/elasticsearch_settings.tsx [hunk 3]
```diff
@@ -35,7 +54,7 @@ export default class ElasticsearchSettings extends AdminSettings {
         return config;
     };
 
-    getStateFromConfig(config) {
+    getStateFromConfig(config: AdminConfig) {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/elasticsearch_settings.tsx:57]: this is actually `(config: Props['config'])`

File: webapp/channels/src/components/admin_console/elasticsearch_settings.tsx [hunk 4]
```diff
@@ -95,7 +114,7 @@ export default class ElasticsearchSettings extends AdminSettings {
         return this.state.canSave;
     };
 
-    doTestConfig = (success, error) => {
+    doTestConfig = (success: (arg0?: any) => void, error: (arg0: any) => void): void => {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/elasticsearch_settings.tsx:117]: can you change the type to 
```
(
        success: (data?: any) => void,
        error: (error: any) => void
    ) => void;
```

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Comment by M-ZubairAhmed** [comment]
@karan2704 please resolve conflicts

**Pull request closed without merging**